x-pack/plugins/security_solution/public/detections/components/rules/step_rule_actions/index.tsx

```diff
@@ -148,7 +149,11 @@ const StepRuleActionsComponent: FC<StepRuleActionsProps> = ({
   ) : (
     <>
       <StepContentWrapper addPadding={!isUpdateView}>
-        <Form form={form} data-test-subj="stepRuleActions">
+        <Form
+          form={form}
+          style={{ display: application.capabilities.actions.show ? 'block' : 'none' }}
+          data-test-subj="stepRuleActions"
+        >
```

Comment:
If things begin to feel or get messy in here, feel ok to drop all the boolean logic weirdness with expressions. That is meant to be a helper for when things look very templatey and nice in React but as time marches on along with permission booleans and things begin getting complex where we have branching logic within branching logic and things look like a bunch of && and ! mashed together we can refactor out the expressions and do regular:

`if/else if/else if/else`

or if we want to we can do the regular:
`if (condition) return this chunk of code early/if (second condition) return this chunk of code ...`  

Since that will allow us to execute more than one block of code within brackets compared to excessive and hard to reason about expressions within expressions and React doing odd things like adding {false/null/undefined<></>} to its tree to later parse out and remove. 

I would avoid having elements with `block` vs `none` on them compared to just not displaying the component altogether based on conditional logic. Usually it is better to omit an HTML element altogether and DOM weight unless we have a bunch of DOM elements that are constantly being added and removed in which case doing a  shown vs. being hidden is better (It will then prevent reflows from happening).

Already looking at this logic if it was me writing the code I would drop all the boolean template expressions and just use normal if/else if logic rather than mixing all these booleans together and trying to get them all correct.

Already we have these that cause us to keep a lot of state in our head:

* isReadOnlyView
* isUpdateView
* application.capabilities.actions.show

And the way the logic flows it's not clear branches that show relationships between the booleans but rather the booleans look rather disjointed from each other and nested within each other.

Reply:
Other tips is there is an `unset` keyword if you're going this route you can utilize maybe instead of assuming that your form is always going to be a `block`:
https://developer.mozilla.org/en-US/docs/Web/CSS/unset#:~:text=The%20unset%20CSS%20keyword%20resets,its%20initial%20value%20if%20not.

Also you might be able to use destructoring to avoid assuming the block?

```ts
...{ application.capabilities.actions.show ? { display: 'block' } : {} }
```

But you will always render the form there when you have data as you introduce a new object but I don't think that's going to be a big deal.

Reply:
I agree with @FrankHassanabad's comment here. We could just do something like this to avoid having to use style `block` or `none`, right?
```
{ application.capabilities.actions.show && (
   <Form ...>...</Form>
)}

Reply:
The result of not rendering the actions step form is having to increase the complexity of the state of the whole rule creation form. The state of the rule creation form depends on pulling a default value from this and when the actions step form is not rendered, the rule creation form blows up. I'm going to look into using the `GhostFormField` and maybe rendering that with the `throttle` default somehow so that I don't need to mess with state.

Also a good thing to note, after talking with @rylnd I learned that we are submitting the entire form on **edit**. This implies that an analyst with no privileges on actions goes to edit a rule created by another analyst **with** actions privileges and they will be greeted with this upon "save" in the "edit rules" page.

<img width="419" alt="Screen Shot 2020-10-01 at 5 11 30 PM" src="https://user-images.githubusercontent.com/915763/94934709-15abfe00-049a-11eb-8f40-80d6baf78f6d.png">